### ISSUE TITLE
Source Mailchimp: Expand API support

### ISSUE THREAD

**Issue by sherifnada** [airbytehq/airbyte#1936]
## Tell us about the problem you're trying to solve
I'd like to be able to read more of the resources exposed by the Mailchimp API. Here are the resources I'd like to read: 

* Segment
* Segment_Member
* List (exists + incremental)
* Member
* Campaign (exists + incremental)
* Campaign Recepient Activity
* Campaign Recepient
* Campaign Variate combination
* Unsubscribe
* Automation Recepient
* Automation Recepient Activity
* Email activity (described in #1693) (should be named `reports_email_activity` since there are other reports we'll add in the future)

These resources should support incremental sync where possible. 

It may make sense to structure this into separate issues/PRs where we add some endpoints + their incremental support, then add some more etc.. rather than add all resources in one go. 

## Describe the alternative you’ve considered or used
Fivetran, Stitchdata, and tap-mailchimp by Singer. 

|Stream | Parent | Incremental | Has data | Has POST API | Notes|
|----|----|----|----|----|----|
|Segment | List | yes | | yes | |
|Segment_Member | List/Segment | no | | yes | |
|List | | yes | | yes | |
|List Member | List | yes | | yes | |
|Campaign | | no | | yes | |
|Campaign Recipient Activity? List Member Activity| List/Subscriber| | | | |

Campaign Recipient? |
Campaign Variate combination? |
Unsubscribe | Campaign | no | |
Automation | | no | | yes |
Automation Recipient? |
Automation Recipient Activity? |
Email activity | Campaign | yes | | | need to use batch API

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368211138838) by [Unito](https://www.unito.io)

**Comment by eugene-kulak** [comment]
I propose to go in batches:

- [ ] ticket to add only parent streams with incremental support
- [ ] ticket to add nested streams
- [ ] ticket to add streams that require batch API

Streams that have the column `Has POST API` empty will require extra time to figure out if it is possible to create test data.

**Comment by eugene-kulak** [comment]
I can't find some stream, see the list of APIs https://github.com/VingtCinq/python-mailchimp#api-structure

**Comment by sherifnada** [comment]
Guessing that this is on the larger side, needs to be split up as eugene mentioned.

**Comment by sherifnada** [comment]
@sherifnada Needs to resolve the questions about streams to support. Currently blocked on me.

**Comment by sherifnada** [comment]
@cgardens you're right, I missed this. closing

**Comment by sean-mckeever** [comment]
To make sure duplicative work doesn't happen, I am biting off a very small chunk of this. My plan is to add one new stream (for now) called List Members, which is mentioned in the original request. I need this specific stream for a use case at my company.

It's my first attempt at doing any airbyte development so fingers crossed I can actually make it happen!

**Comment by pkillarney91** [comment]
Is List Members available yet?

**Comment by misteryeo** [comment]
Closing this in favor of having individual issues for individual streams. 

@pkillarney91 we haven't made List Members available yet as we haven't seen others request this. If you're still interested in having this available, please create a specific issue and we'll make sure to track the users upvoting that request and prioritize accordingly. We'd also love for you to contribute if you're interested!
